src/altinn-app-frontend/src/components/presentation/NavBar.test.tsx

```diff
@@ -1,26 +1,70 @@
-import { mount } from 'enzyme';
 import React from 'react';
-import NavBar from './NavBar';
+import NavBar, { INavBarProps } from './NavBar';
+import { render, screen } from '@testing-library/react';
+import { getLanguageFromCode } from 'altinn-shared/language';
+import userEvent from '@testing-library/user-event';
+import { IAppLanguage } from 'altinn-shared/types';
+
+const renderNavBar = (props?: Partial<INavBarProps>) => {
+  const mockClose = jest.fn();
+  const mockBack = jest.fn();
+  const mockAppLanguageChange = jest.fn();
+  const mockAppLanguages: IAppLanguage[] = [
+    {
+      language: 'nb',
+    },
+    {
+      language: 'en',
+    },
+  ];
+  render(
+    <NavBar
+      language={getLanguageFromCode('nb')}
+      handleClose={mockClose}
+      handleBack={mockBack}
+      appLanguages={mockAppLanguages}
+      onAppLanguageChange={mockAppLanguageChange}
+      selectedAppLanguage={'nb'}
+      {...props}
+    />,
+  );
+
+  return { mockClose, mockBack, mockAppLanguageChange };
+};
 
 describe('components/presentation/NavBar.tsx', () => {
-  it('should render back button by default', () => {
-    const wrapper = mount(
-      <NavBar language={{}} handleClose={null} handleBack={null} />,
-    );
-    expect(wrapper.find('div.a-modal-navbar')).toHaveLength(1);
-    expect(wrapper.find('button.a-modal-close')).toHaveLength(1);
+  it('should render close button', async () => {
+    const { mockClose } = renderNavBar();
+    const closeButton = screen.getByRole('button', { name: /Lukk Skjema/i });
+    await userEvent.click(closeButton);
+    expect(mockClose).toHaveBeenCalled();
+  });
+
+  it('should hide close button', () => {
+    renderNavBar({ hideCloseButton: true });
+    expect(screen.queryByRole('button', { name: /Lukk Skjema/i })).toBeNull();
   });
 
-  it('should not render back button when hideBackButton is supplied', () => {
-    const wrapper = mount(
-      <NavBar
-        language={{}}
-        handleClose={null}
-        handleBack={null}
-        hideCloseButton={true}
-      />,
-    );
-    expect(wrapper.find('div.a-modal-navbar')).toHaveLength(1);
-    expect(wrapper.find('button.a-modal-close')).toHaveLength(0);
+  it('should render back button', async () => {
+    const { mockBack } = renderNavBar({ showBackArrow: true });
+    const backButton = screen.getByRole('button', { name: /Tilbake/i });
+    await userEvent.click(backButton);
+    expect(mockBack).toHaveBeenCalled();
+  });
+  it('should not render back button', async () => {
+    renderNavBar();
+    expect(screen.queryByRole('button', { name: /Tilbake/i })).toBeNull();
```

Comment:
Could this query be less explicit?
It doesn't really protect against if there for instance is a button without  ´/Tilbake/i´ in the name prop.
If I were to change anything with the button this would still pass.

Reply:
Would it be better if I define a variable `const backButtonNamRegex = /Tilbake/i` at the top of the test file and then use this variable in all the Role queries?

Or if you have any other suggestions, I would be glad to hear :D

Reply:
What I mean is that you are checking for the button to not exist. Therefore the button-text should probably not be part of the query.
Maybe even counting the amount of buttons, would be a more useful check here?
If the base language changed the tests would still pass, because there never would be a button that matched the query.
The test in it self is not really protecting against any changes.